blackbird_python/blackbird/listener.py

```diff
@@ -447,26 +445,39 @@ def exitStatement(self, ctx: blackbirdParser.StatementContext):
             self._program._operations.append(operation)
 
     def enterForloop(self, ctx: blackbirdParser.ForloopContext):
-        self._infor = True
+        self._in_for = True
 
     def exitForloop(self, ctx: blackbirdParser.ForloopContext):
-        self._infor = False
+        self._in_for = False
 
-        if ctx.INT():
-            _FORVAR = range(int(ctx.INT().getText()))
+        if ctx.rangeval():
+            for_var = range(*[
+                int(c.getText()) for c in ctx.rangeval().getChildren()
+                if c.getText() != ":"
+            ])
```

Comment:
This is quite nice, since I suppose
```python
for i in 5:10
```
and 
```python
for i in 5:10:2
```
will both work equally well, since `range()` only requires 2 arguments, but can accept 3.

Reply:
Yep, that would be the case. :slightly_smiling_face: